compiler/src/iree/compiler/Dialect/Flow/Transforms/RaiseSpecialOps.cpp

```
@@ -27,6 +28,170 @@ namespace Flow {
+      ++currOutDim;
+      continue;
+    }
+    if (expr.isa<AffineConstantExpr>() &&
+        expr.cast<AffineConstantExpr>().getValue() == 0) {
+      curr.push_back(idx);
+      continue;
+    }
+    return failure();
+  }
+  if (currOutDim != outputIndexingMap.getNumResults())
+    return failure();
+
+  // Push the remaining to the last reassociation if it exists.
+  if (!curr.empty())
+    reassociation.back().append(curr.begin(), curr.end());
+
+  rewriter.replaceOpWithNewOp<tensor::CollapseShapeOp>(linalgOp, input,
+                                                       reassociation);
+  return success();
+}
+
+static LogicalResult tryRaiseToView(linalg::GenericOp linalgOp,
+                                    RewriterBase &rewriter) {
+  if (!linalgOp.hasTensorSemantics())
+    return failure();
+
+  // Assume there is only 1 input, and 1 init tensor.
+  if (linalgOp.getNumDpsInputs() != 1 || linalgOp.getNumDpsInits() != 1)
+    return failure();
+  OpOperand *inputOperand = linalgOp.getDpsInputOperand(0);
+  OpOperand *outputOperand = linalgOp.getDpsInitOperand(0);
+
+  // Check if linalg.yield yields a block arguement.
+  auto yieldOp = dyn_cast<linalg::YieldOp>(linalgOp.getBody()->getTerminator());
+  if (!yieldOp)
+    return failure();
+  // Check if the result is a block argument.
+  auto blockArg = dyn_cast<BlockArgument>(yieldOp.getOperand(0));
+  if (!blockArg)
+    return failure();
+  // Check if the block argument is an argument of the linalgOp.
+  if (!llvm::is_contained(linalgOp.getBody()->getArguments(), blockArg))
+    return failure();
+
+  Value input = inputOperand->get();
+  Value output = outputOperand->get();
+  AffineMap inputIndexingMap = linalgOp.getMatchingIndexingMap(inputOperand);
+  AffineMap outputIndexingMap = linalgOp.getMatchingIndexingMap(outputOperand);
+
+  // TODO: Add raise to tensor.extract_slice.
+  // Try raising to tensor.collapse_shape.
+  (void)tryRaiseToCollapseShape(inputIndexingMap, outputIndexingMap, input,
```

return `failure` here instead of ignore it.